forem/forem
# Allow multiple users to fetch all articles from the same RSS feed

## What type of PR is this? (check all applicable)


## Description
Currently when two or more users try to access the same rss feed, one article will only be fetched in one user account because of a unique `feed_source_url` constraint on the articles table. This PR partially resolves that issue.

**Resolved:**
 - All articles from the same feed can be fetched by all accounts having the same feed_url as long as user does not select the option to automatically set the canonical_url of the article to the source url, i.e. articles will be fetched without issue when `feed_mark_canonical = false`.

**Unresolved:**
- [RESOLVED] As mentioned above, if `feed_mark_canonical` is set to true, articles will be fetched by multiple accounts on a first come first serve basis, because of the globally unique canonical url constraint.
- [RESOLVED] The feature to flag  admins as discussed in [#13185 (comment)](https://github.com/forem/forem/issues/13185#issuecomment-873746408) has not been implemented.

**Conflicts**
- Partially unique index was added on feed_source_url and unique index was removed, which is in conflict with merged PR #10208 .
- To solve the canonical_url issue, we would have to remove the uniqueness constraint on it as well, which is in conflict with merged PR #10161
 
## Related Tickets & Documents
- #13185 

## QA Instructions, Screenshots, Recordings

- Even though multiple users can fetch the same articles, only one user should get to publish a particular article to avoid duplicates. When a user tries to publish an article from the feed that has been published before, following error will pop up:
![feed_source_url](https://user-images.githubusercontent.com/59063821/124777487-deeda180-df5d-11eb-93ab-ddf78b5402fd.png)


## Added/updated tests?


## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_
